client/my-sites/checkout/checkout/composite-checkout.js

```diff
@@ -275,6 +268,32 @@ function useRedirectIfCartEmpty( items, redirectUrl ) {
 	}, [ redirectUrl, items, prevItemsLength ] );
 }
 
+function useCountryList( overrideCountryList ) {
+	// Should we fetch the country list from global state?
+	const shouldFetchList = overrideCountryList?.length <= 0;
+
+	const [ countriesList, setCountriesList ] = useState( overrideCountryList );
+
+	const dispatch = useDispatch();
+	const globalCountryList = useSelector( state => getCountries( state, 'payments' ) );
+
+	// Has the global list been populated?
+	const listIsFetched = globalCountryList?.length > 0;
+
+	useEffect( () => {
+		if ( shouldFetchList ) {
+			if ( listIsFetched ) {
+				setCountriesList( globalCountryList );
+			} else {
+				debug( 'countries list is empty; dispatching request for data' );
+				dispatch( fetchPaymentCountries() );
+			}
+		}
+	}, [ shouldFetchList, listIsFetched, globalCountryList, setCountriesList, dispatch ] );
```

Comment:
I think you can remove `setCountriesList` from this list because React will guarantee that it is never changed and the linter can tell it's directly from `useState` (if I'm wrong and you get a linter warning, it's fine to leave it).